src/backend/utils/adt/dbsize.c

```diff
@@ -89,21 +89,30 @@ get_size_from_segDBs(const char *cmd)
 	for (i = 0; i < cdb_pgresults.numResults; i++)
 	{
 		Datum		value;
+		ExecStatusType status;
+		int ntuples;
+		int nfields;
+
 		struct pg_result *pgresult = cdb_pgresults.pg_results[i];
 
-		if (PQresultStatus(pgresult) != PGRES_TUPLES_OK)
+		status = PQresultStatus(pgresult);
+		if (status != PGRES_TUPLES_OK)
 		{
 			cdbdisp_clearCdbPgResults(&cdb_pgresults);
 			ereport(ERROR,
 					(errmsg("unexpected result from segment: %d",
-							PQresultStatus(pgresult))));
+							status)));
 		}
```

Comment:
Hi, I'm confused here. It there any different from origin codes?
cdb_pgresults will be cleaned up if  status != PGRES_TUPLES_OK.

Reply:
Addressed, ereport will use pgresult which is freed by cdbdisp_clearCdbPgResults.